### ISSUE TITLE
Infinite recursion (macros)

### ISSUE THREAD

**Issue by danmar** [danmar/simplecpp#84]
This issue is copied from http://trac.cppcheck.net/ticket/8106

stbrowne wrote:

I'll admit ahead of time that this is probably an unlikely case, but I actually did run into this recently in some code.

Here is a minimum reproducible sample:

    int main() {
        #define CONVERT(x) ModifyValue(x);
        #define MyClassOld MyClass
        #define MyClass(x) CONVERT(MyClassOld(x))
    /// @begin auto generated code
        return DoSomething(MyClass(5));
    /// @end auto generated code
        #undef MyClassOld
        #undef MyClass
    }

Basically there was a hack to modify some values in autogenerated code so the autogenerated code would not have to be modified after it was generated every time.

Note the interesting thing (and my workaround) is that if you don't use the CONVERT macro and instead just call ModifyValue directly then it works just fine.

**Comment by danmar** [comment]
this is maybe related with #49 and #58

**Comment by danmar** [comment]
I can't reproduce problems with latest git head. For Cppcheck I can reproduce problems with 1.80 but not with 2.0.
